### PR TITLE
Removed bad cereal API call bug

### PR DESCRIPTION
**Pull request by geoffviola**
Called the correct cereal API function so that the the C string does not get implicitly cast to std::string
Cleaned up code

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,46 +1,22 @@
...
-# include_directories(include ${Boost_INCLUDE_DIR} ${catkin_INCLUDE_DIRS})
-# CATKIN_MIGRATION: removed during catkin migration
-# cmake_minimum_required(VERSION 2.4.6)
 
-# CATKIN_MIGRATION: removed during catkin migration
-# include($ENV{ROS_ROOT}/core/rosbuild/rosbuild.cmake)
+find_package(catkin REQUIRED COMPONENTS roscpp message_generation std_msgs)
 
+catkin_package(
+    CATKIN_DEPENDS roscpp std_msgs
```
Comment (mikepurvis) [inline CMakeLists.txt]: You don't need these if you're not exporting libraries.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,46 +1,22 @@
...
+)
 
-# CATKIN_MIGRATION: removed during catkin migration
-# rosbuild_init()
+include_directories(cereal_port/include)
+include_directories(${catkin_INCLUDE_DIRS})
+add_definitions("-std=c++11 -Wall -Wextra -Werror")
```
Comment (mikepurvis) [inline CMakeLists.txt]: Please use `set_directory_properties` for this, eg: https://github.com/ros/ros_comm/blob/kinetic-devel/clients/roscpp/CMakeLists.txt#L5

**Comment by mikepurvis** [comment]
Overall looks great, thanks for doing this cleanup!

We are not actively using this driver. Would you be interested in maintaining it going forward? If so, I would move it to the ros-drivers org and give you write access so that you could merge PRs, use bloom to create releases, etc.

**Comment by geoffviola** [comment]
> We are not actively using this driver. Would you be interested in maintaining it going forward?

Yes, we are actively using this driver. I'm fine with maintaining it on ros-drivers.

**Comment by mikepurvis** [comment]
Awesome, thanks. Please submit a PR changing the maintainer tag in `package.xml`. I'll move it over in the next day or two.

File: package.xml [hunk 3]
```diff
@@ -2,13 +2,16 @@
...
 
   <license>BSD</license>
 
   <url type="website">None</url>
 
-  <author>Jeff Schmidt</author>
+  <author>
+    Jeff Schmidt
+    Geoffrey Viola
+  </author>
```
Comment (mikepurvis) [inline package.xml]: One tag per author.

**Comment by mikepurvis** [comment]
Looks great, thanks!

**Comment by mikepurvis** [comment]
@geoffviola Thanks for taking this on. You should now have access to the following two repos:

https://github.com/ros-drivers/kvh_drivers
https://github.com/ros-drivers-gbp/kvh_drivers-release

You can use bloom to create releases of this repo for the buildfarm to make available as debian packages.

**Comment by geoffviola** [comment]
That's great. Thanks.

**Pull request merged**